include/maths/CBoostedTree.h

```diff
@@ -166,11 +146,38 @@ class MATHS_EXPORT CBoostedTree final : public CDataFrameRegressionModel {
     std::size_t estimateMemoryUsage(std::size_t numberRows, std::size_t numberColumns) const;
 
 private:
-    class CImpl;
-    using TImplUPtr = std::unique_ptr<CImpl>;
+    using TImplUPtr = std::unique_ptr<CBoostedTreeImpl>;
+
+private:
+    CBoostedTree(std::size_t numberThreads, std::size_t dependentVariable, TLossFunctionUPtr loss);
+
+    //! \name Parameter Setters
+    //@{
+    //! Set the number of folds to use for estimating the generalisation error.
+    CBoostedTree& numberFolds(std::size_t folds);
```

Comment:
Are any of these setters actually called? You call m_Tree->m_Impl->numberFolds(folds) from the factory for example. I think these functions can just be dropped.